### PR TITLE
Fix #107: surface errors correctly, return a promise, no expectations.

### LINKED ISSUE #107: Build is incorrectly reporting success
```
Unhandled rejection AssertionError: expected 'INITIAL TEST VALUE WHICH SHOULD BE OVERRIDDEN' to equal undefined
    at assertEqual (/home/travis/build/matt-kruse/alexa-app/node_modules/chai/lib/chai/core/assertions.js:487:12)
    at ctx.(anonymous function) (/home/travis/build/matt-kruse/alexa-app/node_modules/chai/lib/chai/utils/addMethod.js:41:25)
    at doAsserterAsyncAndAddThen (/home/travis/build/matt-kruse/alexa-app/node_modules/chai-as-promised/lib/chai-as-promised.js:296:33)
    at null.<anonymous> (/home/travis/build/matt-kruse/alexa-app/node_modules/chai-as-promised/lib/chai-as-promised.js:255:21)
    at ctx.(anonymous function) [as equal] (/home/travis/build/matt-kruse/alexa-app/node_modules/chai/lib/chai/utils/overwriteMethod.js:49:33)
    at /home/travis/build/matt-kruse/alexa-app/test/test_alexa_app_session.js:229:55
    at tryCatcher (/home/travis/build/matt-kruse/alexa-app/node_modules/bluebird/js/main/util.js:26:23)
    at Promise._settlePromiseFromHandler (/home/travis/build/matt-kruse/alexa-app/node_modules/bluebird/js/main/promise.js:510:31)
    at Promise._settlePromiseAt (/home/travis/build/matt-kruse/alexa-app/node_modules/bluebird/js/main/promise.js:584:18)
    at Promise._settlePromiseAtPostResolution (/home/travis/build/matt-kruse/alexa-app/node_modules/bluebird/js/main/promise.js:248:10)
    at Async._drainQueue (/home/travis/build/matt-kruse/alexa-app/node_modules/bluebird/js/main/async.js:128:12)
    at Async._drainQueues (/home/travis/build/matt-kruse/alexa-app/node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/travis/build/matt-kruse/alexa-app/node_modules/bluebird/js/main/async.js:15:14)
    at tryOnImmediate (timers.js:534:15)
    at processImmediate [as _immediateCallback] (timers.js:514:5)
```

https://travis-ci.org/matt-kruse/alexa-app/builds/187971131

### PR DESCRIPTION
**Pull request by dblock**

### CODE REVIEW

File: test/test_alexa_app_session.js [hunk 1]
```diff
@@ -224,10 +224,9 @@ describe("Alexa", function() {
           };
         });
 
-        it("session.get(key) should not throw if attribute is not present", function() {
+        it("session.get(key) should throw if attribute is not present", function() {
```
Comment (dblock) [inline test/test_alexa_app_session.js]: This now throws because we don't have a session (`{}.get(...)`). Is that correct?

```js
returnedAttributeValue = req.getSession().get("AttributeWhichDoesNotExist");
```

Comment (trayburn) [inline test/test_alexa_app_session.js]: Ok, back from my errands now, sorry for the delay.  I do not believe this is correct, I believe the requested behavior in #98 was that we return undefined.  The problem I believe was simply one of how I wrote the test.  I wanted two asserts, one that the `app.request()` call did not throw, the other that the `session.get()` call returned undefined, but by using a `.then()` I mucked with the `Promise`.

Summing up I believe the desired state is:
1. `session.get("unknown_key")` should return undefined
1. The `app.request(data)` call should not throw just because of that missing key

Comment (dblock) [inline test/test_alexa_app_session.js]: But this is in the case of `NO_SESSION`, what am I missing?

**Comment by dblock** [comment]
@trayburn would appreciate a code review of this whole thing here

File: test/test_alexa_app_session.js (hunk 1 above)
Comment (dblock) [inline test/test_alexa_app_session.js]: I believe (2) is checked with my change, but I could be wrong ;)

**Comment by trayburn** [comment]
Review complete, so it gets my +1

**Comment by dblock** [comment]
@trayburn what about that `NO_SESSION` thing above?
